shared/instance.go

```diff
@@ -28,6 +28,12 @@ const (
 // ConfigVolatilePrefix indicates the prefix used for volatile config keys.
 const ConfigVolatilePrefix = "volatile."
 
+// NicConfig contains network interface configuration to be passed into a VM and applied by the agent.
+type NicConfig struct {
```

Comment:
I'm not sure this struct should be in `shared`, why does it need to be consumed outside of LXD (i.e in `lxc` or `client` packages?) and if not it shouldn't be here. Thanks

Reply:
I resolved this prematurely. 

Can you recommend somewhere for this to go please? I have tried `lxd/instance/instance_interface.go` as well as `lxd/device/nic.go` but the agent fails to build in both cases due to build constraints. (I think one due to disabling CGO and the other being an explicit `!agent` build tag in the `db` package).

Reply:
`github.com/lxc/lxd/lxd/device/config` seems like a good place as it represents NIC device config.